client/my-sites/people/invite-people/index.jsx

```diff
@@ -286,113 +295,183 @@ const InvitePeople = React.createClass( {
 				onClick={ this.onClickRoleExplanation }
 			>
-				{ this.translate( 'Learn more about roles' ) }
+				{ translate( 'Learn more about roles' ) }
 			</a>
 		);
 	},
 
+	enableSSO() {
+		this.props.activateModule( this.props.siteId, 'sso' );
+	},
+
+	renderInviteForm() {
+		const {
+			site,
+			translate,
+			needsVerification,
+			isJetpack,
+			showSSONotice
+		} = this.props;
+
+		const inviteForm = (
+			<Card>
+				<EmailVerificationGate>
+					<form onSubmit={ this.submitForm } >
+						<div role="group" className="invite-people__token-field-wrapper">
+							<FormLabel>{ translate( 'Usernames or Emails' ) }</FormLabel>
+							<TokenField
+								isBorderless
+								tokenizeOnSpace
+								autoCapitalize="none"
+								autoComplete="off"
+								maxLength={ 10 }
+								value={ this.getTokensWithStatus() }
+								onChange={ this.onTokensChange }
+								onFocus={ this.onFocusTokenField }
+								disabled={ this.state.sendingInvites } />
+							<FormSettingExplanation>
+								{ translate(
+									'Invite up to 10 email addresses and/or WordPress.com usernames. ' +
+									'Those needing a username will be sent instructions on how to create one.'
+								) }
+							</FormSettingExplanation>
+						</div>
+
+						<RoleSelect
+							id="role"
+							name="role"
+							includeFollower
+							siteId={ this.props.siteId }
+							onChange={ this.onRoleChange }
+							onFocus={ this.onFocusRoleSelect }
+							value={ this.state.role }
+							disabled={ this.state.sendingInvites }
+							explanation={ this.renderRoleExplanation() }
+							/>
+
+						<FormFieldset>
+							<FormLabel htmlFor="message">{ translate( 'Custom Message' ) }</FormLabel>
+							<CountedTextarea
+								name="message"
+								id="message"
+								showRemainingCharacters
+								maxLength={ 500 }
+								acceptableLength={ 500 }
+								onChange={ this.onMessageChange }
+								onFocus={ this.onFocusCustomMessage }
+								value={ this.state.message }
+								disabled={ this.state.sendingInvites } />
+							<FormSettingExplanation>
+								{ translate(
+									'(Optional) You can enter a custom message of up to 500 characters ' +
+									'that will be included in the invitation to the user(s).'
+								) }
+							</FormSettingExplanation>
+						</FormFieldset>
+
+						<FormButton disabled={ this.isSubmitDisabled() } onClick={ this.onClickSendInvites } >
+							{ translate(
+								'Send Invitation',
+								'Send Invitations', {
+									count: this.state.usernamesOrEmails.length || 1,
+									context: 'Button label'
+								}
+							) }
+						</FormButton>
+					</form>
+				</EmailVerificationGate>
+			</Card>
+		);
+
+		// Return early for WPCOM or needs verification
+		if ( ! site || ! isJetpack || needsVerification ) {
+			return inviteForm;
+		}
+
+		const jetpackVersion = get( site, 'options.jetpack_version', 0 );
+		if ( versionCompare( jetpackVersion, '4.9-alpha', '<' ) ) {
```

Comment:
I think we'll actually need to update this to `4.10-alpha` now that we missed the release. But, yes, we'd need to run whatever version includes the new endpoints in Jetpack.